storage/remote/queue_manager.go

```diff
@@ -33,6 +33,33 @@ const (
 	dropped = "dropped"
 )
 
+var (
```

Comment:
I'm not sure that the metrics structure needs many updates. You already return a different name for each storage client, and there's a queue for every client that puts that name into its ConstLabels, so that should be sufficient?